lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/util/SystemArguments.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.base.util;
+
+import com.google.common.collect.Sets;
+
+import java.lang.management.ManagementFactory;
+import java.util.Optional;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static java.util.stream.Collectors.toSet;
+
+public final class SystemArguments
```

Comment:
I think this is useful to have in plugin-toolkit - if you disagree I'll move it to BigQuery plugin itself.

Reply:
Let’s keep it in the bigquery module. It should be rare to need this and until run into other cases we don’t need to make it reusable.